contracts/ibc-callbacks/src/contract.rs

```diff
@@ -0,0 +1,174 @@
+use cosmwasm_std::{
+    entry_point, to_json_binary, to_json_string, Addr, Binary, Deps, DepsMut, Empty, Env,
+    IbcBasicResponse, IbcMsg, IbcPacketAckMsg, IbcPacketTimeoutMsg, IbcTimeout, MessageInfo,
+    Response, StdError, StdResult, Uint64,
+};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::msg::{ExecuteMsg, QueryMsg};
+use crate::state::{load_stats, save_stats, CallbackStats};
+
+#[entry_point]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _msg: Empty,
+) -> StdResult<Response> {
+    // initialize counts
+    let counts = CallbackStats::default();
+    save_stats(deps.storage, &counts)?;
+
+    Ok(Response::new().add_attribute("action", "instantiate"))
+}
+
+/// Sends an `IbcMsg::Transfer` to the given `to_address` on the given `channel_id`.
+#[entry_point]
+pub fn execute(
+    _deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> StdResult<Response> {
+    let transfer_msg = match &*info.funds {
+        [coin] if !coin.amount.is_zero() => IbcMsg::Transfer {
+            to_address: msg.to_address,
+            timeout: IbcTimeout::with_timestamp(env.block.time.plus_seconds(5)), // TODO: how much time?
```

Comment:
This should be configurable in the `ExecuteMsg`. If you make it low, like 2s, chances are very high it will time out. Of you make it 200s, chances are high it will be acknowledged. This ways a caller can test both scenarios.